### ISSUE TITLE
Pretrained models' training condition

### ISSUE THREAD

**Issue by kwea123** [NVIDIA/semantic-segmentation#25]
What is the training condition of your `cityscapes_best.pth`?
Is it
1. Pretrained on Mapillary?
2. Use label relaxation loss?
3. Use sdc-aug label propagation?

I want to do comparison with my models. Thank you.

**Comment by bryanyzhu** [comment]
I don't have it right now, but I can train one for you. I will let you know once it is done.

**Comment by kwea123** [comment]
Thank you. You are so kind.

**Comment by bryanyzhu** [comment]
@kwea123  The training is done, I did a PR at [here](https://github.com/NVIDIA/semantic-segmentation/pull/26). You can try the weights there.

**Comment by kwea123** [comment]
Thank you!

**Comment by kwea123** [comment]
Can you also provide the model pretrained on Mapillary only? Thanks for your kindness.

**Comment by bryanyzhu** [comment]
Due to license issue, we are not able to release the Mapillary pre-trained models. You can follow our readme to train your own model. It should reproduce, although I realize that training on Mapillary takes some time. Sorry for the inconvenience.

**Comment by kwea123** [comment]
Thanks for the reply. In this case, can you just share the results (the IoU per class) to allow for comparison? It is also desirable if you can share results for both seresnext and widerresnet.

**Comment by bryanyzhu** [comment]
@karansapra Do you have the number? Or the training log of Mapillary models before? Thank you.

**Comment by karansapra** [comment]
I remember the mIOU was close to 50.03 but this resized, padded and drop last IOU. Let me double-check and get back to you on this.

**Comment by bryanyzhu** [comment]
@kwea123 Just let you know, I'm retraining a widerresnet model on Mapillary to get the per-class IoU. I'll report the number here once the training is done (several days).

**Comment by bryanyzhu** [comment]
@kwea123 Sorry for the long wait, I have re-trained both models on Mapillary and attach the results here. Note that, these results are obtained on validation set during training. As @karansapra have said, this is the resized image, padded and drop_last=True mIoU, it doesn't involve fancy testing strategies such as sliding, flipping and multi-scale. So the numbers here should be 1~2% lower than numbers reported in other papers, but you can use these for comparison. 

```
SE-ResNeXt50 on Mapillary validation set:

10-02 22:37:15.419 IoU:
10-02 22:37:15.419 label_id      label    iU    Precision Recall TP     FP    FN
10-02 22:37:15.419  0                0    0.20   0.00      0.72   0.00   508.42    0.40
10-02 22:37:15.419  1                1   37.78   0.42      0.80   0.00    1.40    0.25
10-02 22:37:15.419  2                2   62.03   0.77      0.76   0.59    0.29    0.32
10-02 22:37:15.420  3                3   61.26   0.75      0.77   0.75    0.33    0.30
10-02 22:37:15.420  4                4   67.20   0.80      0.81   0.15    0.26    0.23
10-02 22:37:15.420  5                5   56.91   0.68      0.78   0.22    0.48    0.28
10-02 22:37:15.420  6                6   52.70   0.66      0.72   0.46    0.50    0.39
10-02 22:37:15.420  7                7   35.96   0.43      0.68   0.19    1.31    0.47
10-02 22:37:15.420  8                8   27.97   0.34      0.62   0.10    1.95    0.62
10-02 22:37:15.420  9                9   18.54   0.24      0.45   0.02    3.17    1.22
10-02 22:37:15.420 10               10   16.95   0.22      0.44   0.05    3.60    1.30
10-02 22:37:15.420 11               11   28.97   0.30      0.90   0.18    2.34    0.11
10-02 22:37:15.420 12               12   56.70   0.67      0.78   0.07    0.48    0.28
10-02 22:37:15.421 13               13   88.91   0.96      0.92  20.07    0.04    0.08
10-02 22:37:15.421 14               14   39.46   0.49      0.68   0.10    1.06    0.48
10-02 22:37:15.421 15               15   68.21   0.85      0.77   2.69    0.17    0.29
10-02 22:37:15.421 16               16   72.64   0.83      0.86   0.57    0.21    0.17
10-02 22:37:15.421 17               17   88.69   0.95      0.93  10.27    0.05    0.07
10-02 22:37:15.421 18               18   26.12   0.28      0.82   0.02    2.61    0.22
10-02 22:37:15.421 19               19   78.04   0.90      0.86   0.26    0.11    0.17
10-02 22:37:15.421 20               20   63.42   0.76      0.79   0.03    0.32    0.26
10-02 22:37:15.421 21               21   66.87   0.80      0.81   0.02    0.26    0.24
10-02 22:37:15.421 22               22    1.90   0.02      0.11   0.00   43.30    8.27
10-02 22:37:15.422 23               23   71.47   0.80      0.87   0.68    0.25    0.15
10-02 22:37:15.422 24               24   64.14   0.75      0.81   1.10    0.33    0.23
10-02 22:37:15.422 25               25   50.44   0.58      0.79   0.13    0.72    0.26
10-02 22:37:15.422 26               26   19.42   0.22      0.63   0.00    3.57    0.58
10-02 22:37:15.422 27               27   98.34   0.99      0.99  31.74    0.01    0.01
10-02 22:37:15.422 28               28   79.76   0.89      0.89   0.31    0.12    0.13
10-02 22:37:15.422 29               29   69.54   0.81      0.83   0.79    0.24    0.20
10-02 22:37:15.422 30               30   90.69   0.96      0.94  13.30    0.04    0.06
10-02 22:37:15.422 31               31   79.14   0.82      0.95   0.06    0.21    0.05
10-02 22:37:15.422 32               32   40.45   0.50      0.68   0.03    0.99    0.48
10-02 22:37:15.423 33               33   46.73   0.58      0.71   0.01    0.73    0.41
10-02 22:37:15.423 34               34   16.66   0.27      0.31   0.00    2.76    2.24
10-02 22:37:15.423 35               35   52.32   0.66      0.71   0.36    0.51    0.40
10-02 22:37:15.423 36               36   42.95   0.54      0.67   0.02    0.84    0.49
10-02 22:37:15.423 37               37   13.31   0.16      0.46   0.00    5.34    1.18
10-02 22:37:15.423 38               38   63.19   0.72      0.84   0.00    0.39    0.19
10-02 22:37:15.423 39               39   46.49   0.57      0.72   0.03    0.77    0.38
10-02 22:37:15.423 40               40   11.50   0.14      0.41   0.00    6.26    1.44
10-02 22:37:15.423 41               41   54.53   0.65      0.77   0.04    0.53    0.30
10-02 22:37:15.423 42               42    5.16   0.07      0.16   0.00   13.18    5.19
10-02 22:37:15.424 43               43   13.57   0.18      0.36   0.00    4.55    1.81
10-02 22:37:15.424 44               44   54.21   0.64      0.77   0.04    0.55    0.29
10-02 22:37:15.424 45               45   56.68   0.72      0.73   0.61    0.40    0.37
10-02 22:37:15.424 46               46   54.32   0.65      0.77   0.07    0.54    0.30
10-02 22:37:15.424 47               47   56.88   0.72      0.73   0.33    0.39    0.36
10-02 22:37:15.424 48               48   73.24   0.84      0.85   0.16    0.19    0.18
10-02 22:37:15.424 49               49   51.00   0.65      0.71   0.05    0.55    0.42
10-02 22:37:15.424 50               50   77.18   0.87      0.87   0.47    0.15    0.15
10-02 22:37:15.424 51               51   63.90   0.78      0.78   0.04    0.29    0.27
10-02 22:37:15.424 52               52   67.62   0.81      0.80   0.04    0.23    0.25
10-02 22:37:15.425 53               53   54.83   0.69      0.73   0.00    0.45    0.37
10-02 22:37:15.425 54               54   84.43   0.91      0.92   0.20    0.10    0.08
10-02 22:37:15.425 55               55   92.01   0.97      0.94   3.04    0.03    0.06
10-02 22:37:15.425 56               56    0.31   0.00      0.03   0.00   286.31   37.24
10-02 22:37:15.425 57               57   62.94   0.77      0.77   0.04    0.29    0.30
10-02 22:37:15.425 58               58   60.78   0.68      0.85   0.02    0.47    0.17
10-02 22:37:15.425 59               59   19.65   0.25      0.49   0.01    3.03    1.06
10-02 22:37:15.425 60               60    9.35   0.10      0.52   0.00    8.76    0.94
10-02 22:37:15.425 61               61   73.97   0.82      0.88   0.29    0.22    0.14
10-02 22:37:15.425 62               62   37.74   0.46      0.67   0.00    1.17    0.48
10-02 22:37:15.426 63               63    8.51   0.39      0.10   0.01    1.56    9.19
10-02 22:37:15.426 64               64   92.34   0.96      0.96   2.00    0.04    0.04
10-02 22:37:15.426 mean 0.5075579285621643
10-02 22:37:18.102 -----------------------------------------------------------------------------------------------------------
10-02 22:37:18.122 [epoch 171], [val loss 0.24626], [acc 0.92842], [acc_cls 0.59483], [mean_iu 0.50756], [fwavacc 0.87452]
10-02 22:37:18.122 best record: [val loss 0.24626], [acc 0.92842], [acc_cls 0.59483], [mean_iu 0.50756], [fwavacc 0.87452], [epoch 171]

```

```
WideResNet38 on Mapillary validation set:

09-30 04:51:25.332 IoU:
09-30 04:51:25.332 label_id      label    iU    Precision Recall TP     FP    FN
09-30 04:51:25.333  0                0    0.07   0.00      0.27   0.00   1511.59    2.76
09-30 04:51:25.333  1                1   23.69   0.26      0.69   0.00    2.78    0.44
09-30 04:51:25.333  2                2   61.64   0.80      0.73   0.60    0.26    0.37
09-30 04:51:25.333  3                3   61.23   0.77      0.75   0.77    0.30    0.34
09-30 04:51:25.333  4                4   66.03   0.80      0.79   0.15    0.25    0.27
09-30 04:51:25.333  5                5   55.13   0.66      0.77   0.21    0.51    0.30
09-30 04:51:25.333  6                6   52.97   0.66      0.73   0.45    0.52    0.37
09-30 04:51:25.333  7                7   34.20   0.41      0.67   0.18    1.44    0.48
09-30 04:51:25.333  8                8   27.72   0.32      0.67   0.10    2.12    0.49
09-30 04:51:25.334  9                9   21.03   0.28      0.45   0.03    2.53    1.22
09-30 04:51:25.334 10               10   18.18   0.22      0.50   0.05    3.51    0.99
09-30 04:51:25.334 11               11   34.06   0.40      0.69   0.24    1.48    0.45
09-30 04:51:25.334 12               12   53.33   0.70      0.69   0.07    0.43    0.45
09-30 04:51:25.334 13               13   88.36   0.95      0.92  19.94    0.05    0.08
09-30 04:51:25.334 14               14   46.10   0.57      0.70   0.12    0.74    0.43
09-30 04:51:25.334 15               15   66.98   0.81      0.80   2.55    0.24    0.26
09-30 04:51:25.334 16               16   74.99   0.87      0.84   0.60    0.15    0.18
09-30 04:51:25.334 17               17   88.70   0.94      0.94  10.18    0.06    0.07
09-30 04:51:25.334 18               18   42.86   0.48      0.81   0.03    1.10    0.23
09-30 04:51:25.335 19               19   76.35   0.89      0.85   0.25    0.13    0.18
09-30 04:51:25.335 20               20   56.84   0.67      0.79   0.02    0.49    0.27
09-30 04:51:25.335 21               21   63.87   0.79      0.77   0.02    0.27    0.30
09-30 04:51:25.335 22               22    8.35   0.10      0.35   0.00    9.13    1.84
09-30 04:51:25.335 23               23   69.49   0.80      0.84   0.68    0.25    0.19
09-30 04:51:25.335 24               24   62.38   0.76      0.77   1.11    0.31    0.29
09-30 04:51:25.335 25               25   49.77   0.57      0.79   0.13    0.74    0.27
09-30 04:51:25.335 26               26   10.40   0.26      0.15   0.00    2.87    5.74
09-30 04:51:25.335 27               27   98.23   0.99      0.99  31.72    0.01    0.01
09-30 04:51:25.335 28               28   75.68   0.92      0.81   0.32    0.09    0.23
09-30 04:51:25.336 29               29   69.73   0.82      0.82   0.80    0.22    0.21
09-30 04:51:25.336 30               30   90.54   0.96      0.94  13.27    0.04    0.06
09-30 04:51:25.336 31               31   78.26   0.81      0.96   0.06    0.24    0.04
09-30 04:51:25.336 32               32   36.31   0.46      0.64   0.03    1.19    0.56
09-30 04:51:25.336 33               33   44.32   0.60      0.62   0.01    0.65    0.60
09-30 04:51:25.336 34               34   13.65   0.21      0.28   0.00    3.76    2.56
09-30 04:51:25.336 35               35   50.22   0.66      0.68   0.36    0.52    0.48
09-30 04:51:25.336 36               36   39.48   0.52      0.63   0.02    0.94    0.60
09-30 04:51:25.336 37               37    6.79   0.07      0.42   0.00   12.36    1.36
09-30 04:51:25.336 38               38   54.21   0.66      0.76   0.00    0.52    0.32
09-30 04:51:25.337 39               39   42.05   0.55      0.64   0.03    0.81    0.57
09-30 04:51:25.337 40               40    6.33   0.10      0.15   0.00    9.31    5.49
09-30 04:51:25.337 41               41   55.71   0.70      0.73   0.04    0.43    0.37
09-30 04:51:25.337 42               42    8.57   0.12      0.25   0.00    7.59    3.08
09-30 04:51:25.337 43               43    7.92   0.11      0.24   0.00    8.46    3.16
09-30 04:51:25.337 44               44   53.21   0.67      0.73   0.04    0.50    0.38
09-30 04:51:25.337 45               45   56.05   0.72      0.72   0.62    0.39    0.39
09-30 04:51:25.337 46               46   52.74   0.64      0.75   0.07    0.57    0.33
09-30 04:51:25.337 47               47   55.82   0.70      0.73   0.32    0.42    0.37
09-30 04:51:25.338 48               48   71.79   0.86      0.82   0.16    0.17    0.22
09-30 04:51:25.338 49               49   49.53   0.66      0.66   0.05    0.51    0.51
09-30 04:51:25.338 50               50   75.93   0.87      0.86   0.47    0.15    0.16
09-30 04:51:25.338 51               51   58.84   0.77      0.71   0.04    0.29    0.41
09-30 04:51:25.338 52               52   64.81   0.79      0.78   0.04    0.27    0.28
09-30 04:51:25.338 53               53   40.79   0.53      0.64   0.00    0.89    0.56
09-30 04:51:25.338 54               54   80.53   0.86      0.93   0.19    0.16    0.08
09-30 04:51:25.338 55               55   91.46   0.97      0.94   3.04    0.03    0.06
09-30 04:51:25.338 56               56   16.36   0.23      0.35   0.00    3.27    1.85
09-30 04:51:25.338 57               57   60.62   0.76      0.75   0.04    0.32    0.33
09-30 04:51:25.339 58               58   54.87   0.64      0.79   0.02    0.56    0.26
09-30 04:51:25.339 59               59   16.78   0.23      0.37   0.01    3.26    1.70
09-30 04:51:25.339 60               60   18.73   0.30      0.33   0.00    2.28    2.06
09-30 04:51:25.339 61               61   68.98   0.81      0.83   0.29    0.24    0.21
09-30 04:51:25.339 62               62   21.46   0.28      0.48   0.00    2.60    1.06
09-30 04:51:25.339 63               63    6.23   0.39      0.07   0.01    1.58   13.46
09-30 04:51:25.339 64               64   91.52   0.96      0.95   2.00    0.04    0.05
09-30 04:51:25.339 mean 0.4922703802585602
09-30 04:51:37.517 -----------------------------------------------------------------------------------------------------------
09-30 04:51:37.543 [epoch 174], [val loss 0.25166], [acc 0.92571], [acc_cls 0.59429], [mean_iu 0.49227], [fwavacc 0.87120]
09-30 04:51:37.543 best record: [val loss 0.25166], [acc 0.92571], [acc_cls 0.59429], [mean_iu 0.49227], [fwavacc 0.87120], [epoch 174]

```

Another thing to note is that, each run of the training will give you a little bit different result. This is common for semantic segmentation problem. For example, the Mapillary model in WideResNet bachbone we used has 50.03 mIoU, a little bit higher than this run 49.23. @karansapra Any luck finding the training log?  

Right now, our evaluation script doesn't support evaluation on Mapillary. We will work on that, and hopefully provide you more accurate numbers in the future.

**Comment by kwea123** [comment]
Thank you, this helps a lot!

**Comment by karansapra** [comment]
Sorry no training log on this, just the weights and command. But this looks right as Yi mentioned. The key is having a good backbone, we have worked with 49.01 on Mapillary and it works fine :)

**Comment by kwea123** [comment]
I was worried when I only got 34% Miou on the mapillary commercial version. But since it has 100 classes and research only has 65 classes, this corresponds to 34*100/65=52% actually. I'm relieved.

**Comment by bryanyzhu** [comment]
Interesting! I haven't played with commercial version before, and I didn't see anyone reporting the number as well. 34% seems low, but maybe it is a challenging dataset. Thank you for sharing the info. 

Start with some visualization will help diagnosing the problem. You will know if the low accuracy is caused by serious class confusion, small classes missing, coarse boundary or sth else. 

Also try a baseline FCN or other popular framework with code (like PSPNet). If their performance is also 30ish, you will be more relieved.

**Comment by kwea123** [comment]
Commercial has the same images as research version, it just additionally labels the unknown part of the research version to make more classes. So basically these additional classes have very low iou. I think it's quite fair to compare roughly the score * the number of classes.

**Comment by bryanyzhu** [comment]
Got it. Yes, it makes sense. With the same number of images, these additional classes are expected to have very low IoU.
